libs/eavmlib/src/network.erl

```diff
@@ -390,10 +370,25 @@ handle_info(Msg, State) ->
     {noreply, State}.
 
 %% @hidden
-terminate(_Reason, _State) ->
+%% Wait for port to be closed
+terminate(_Reason, State) ->
     Ref = make_ref(),
+    Pid = State#state.port,
```

Comment:
Let's call it Port ;-)
Also, the monitor could be set up in `init/1` and used as Ref in `#state`.